ports/mimxrt/hal/pwm_backport.c

```diff
@@ -36,15 +36,15 @@ void PWM_UpdatePwmDutycycle_u16(
 
     // Setup the PWM dutycycle of channel A or B
     if (pwmSignal == kPWM_PwmA) {
-        if (dutyCycle >= 65536) {
+        if (dutyCycle >= PWM_FULL_SCALE) {
```

Comment:
This constant is still defined as 65536 in `pwm_backport.h`.  Does it need changing there?

Reply:
Of course.

Reply:
Updated. Behavior still fine. At duty_u16(32768) the difference between low and high phases with a Teensy 4.x are 700ps (flexpwm) and 400ps (QTMR).

Reply:
Can the hardware have even better symmetry with 50% duty cycle, if you choose the right upper limit for the counter?  If so, it may be possible to get this working with `machine.PWM` by using 65536 for the hardware full scale, but 65535 for the `duty_u16` max, and then scaling it via `x * 65536 / 65535`.  I think this is how the rp2 port works.

Reply:
The difference between the high/low phase is 10% of the PWM clock or about 50% of the CPU clock. It hardly can get much better.

Reply:
Right... picoseconds are very small!  Great work testing it to that degree.